src/directLine.ts

```diff
@@ -202,6 +202,24 @@ export interface Message extends IActivity {
     value?: object
 }
 
+export interface MessageUpdate extends IActivity {
+    type: "update",
+    text?: string,
+    locale?: string,
+    textFormat?: "plain" | "markdown" | "xml",
+    attachmentLayout?: AttachmentLayout,
+    attachments?: Attachment[],
+    entities?: any[],
+    suggestedActions?: { actions: CardAction[], to?: string[] },
+    speak?: string,
+    inputHint?: string,
+    value?: object
+}
+
+export interface MessageDelete extends IActivity {
+    type: "delete"
+}
+
 export interface Typing extends IActivity {
```

Comment:
They are the same, except for the type. I will create an IBaseMessage interface with the common pieces.